### PR TITLE
feat!: add extra port support

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This adds support for adding additional ports to `uds-k3d` for testing with uds packages that may require additional ports to be exposed.

> :warning: **BREAKING CHANGE** This PR removes the passthrough gateway rules for the older login.uds.dev and keycloak.uds.dev entries.

## Related Issue

Fixes #N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: docs/DNS.md
Comment (daveworth) [inline docs/DNS.md:13]: maybe a silly question but ... what changed in this file?

**Review by daveworth — COMMENTED** [review]
reviewed and approved licensing and asked a single question.  I will definitely defer to the UDS Core folks for review

File: docs/DNS.md
Comment (daveworth) [reply docs/DNS.md:13]: I could have sworn the GitHub diff view made that clearer as I often do the same.  Thanks!

File: chart/templates/nginx.yaml
Comment (mjnagel) [inline chart/templates/nginx.yaml:17]: If deleting these we could/should also delete the `upstream backend_passthrough` block below here.

File: chart/templates/nginx.yaml
Comment (mjnagel) [inline chart/templates/nginx.yaml:46]: I may just be missing the connection here, but how would you intend to use `backend_tenant_22` for example? No hosts would point to it, so while it would be exposed I'm not sure it would be routable via a hostname?

Let me know if I'm just missing the wiring here.

**Review by mjnagel — COMMENTED** [review]
Not a huge fan of adding this complexity, especially untested here, but I do see the testing need if trying to enable something like GitLab + SSH.

File: chart/templates/nginx.yaml
Comment (Racer159) [reply chart/templates/nginx.yaml:17]: wilco

File: chart/templates/nginx.yaml
Comment (Racer159) [reply chart/templates/nginx.yaml:46]: https://github.com/defenseunicorns/uds-package-gitlab/pull/196/files#diff-17ed18489a956f326ec0fe4040850c5bc9261d4631fb42da4c52891d74a59180R76
https://github.com/defenseunicorns/uds-package-gitlab/pull/196/files#diff-8cb1344ac0843ca80af6191a6034a36e5cea2f6d9e8f65a734e282210a8892c6

Those are the docs and the ingress wirings from the GitLab PR (stolen from here: https://github.com/defenseunicorns/uds-bundle-software-factory-nutanix/tree/main/packages/additional-manifests/gitlab)

**Comment by Racer159** [comment]
> Not a huge fan of adding this complexity, especially untested here, but I do see the testing need if trying to enable something like GitLab + SSH.


We could add some light testing if we wanted to have a simple nginx app with a loadbalancer in front of it responding over different ports

**Review by mjnagel — APPROVED** [review]

**Review by daveworth — APPROVED** [review]
approving for codeowner pieces per @mjnagel

**Pull request merged**
